docs/src/ratelimit.md

```diff
@@ -0,0 +1,171 @@
+
+# TokenBucket Balanced Mode — Scenario 1 Timeline
+
+Assumptions:
+- Capacity `C = 10`
+- `fillDuration = 1s` (per-token time: 100ms)
+- Start: `t = 0ms`, `budget = 10`, `lastUpdate = 0ms`
+
+Legend:
+- Minted tokens: tokens added by Balanced update at that step (TA)
+- Budget after mint: budget after minting, before the consume at that row
+- Budget after consume: budget left after processing the request at that row
+- LU set?: whether `lastUpdate` changes at that step (reason)
+
+Only request events are listed below (no passive availability checks):
+
+| Time    | Elapsed from LU | Budget (in) | Request tokens | Minted tokens (TA) | Budget after mint | Budget after consume | LU set?                          |
+|---------|------------------|-------------|----------------|--------------------|-------------------|----------------------|-----------------------------------|
+| 0 ms    | n/a              | 10          | 7              | 0                  | 10                | 3                    | yes (consume/full → 0 ms)         |
+| 200 ms  | 200 ms           | 3           | 5              | 2                  | 5                 | 0                    | yes (update → 200 ms)             |
+| 650 ms  | 450 ms           | 0           | 3              | 4                  | 4                 | 1                    | yes (update → 600 ms)             |
+| 1200 ms | 600 ms           | 1           | 6              | 6                  | 7                 | 1                    | yes (update → 1200 ms)            |
+| 1800 ms | 600 ms           | 1           | 5              | 6                  | 7                 | 2                    | yes (update → 1800 ms)            |
+| 2100 ms | 300 ms           | 2           | 10             | 3                  | 5                 | 5 (insufficient)     | yes (update → 2100 ms)            |
+| 2600 ms | 500 ms           | 5           | 10             | 5 (to cap)         | 10 (hit cap)      | 0                    | yes (update hit cap → 2600 ms); yes (consume/full → 2600 ms) |
+
+Notes:
+- When an update would overfill the bucket, it is clamped to capacity and `lastUpdate` is set to the current time; leftover elapsed time is discarded.
+- Consuming from a full bucket sets `lastUpdate` to the consume time (prevents idle-at-full credit banking).
+
+### Consumption Summary (0–3s window)
+
+Per `fillDuration` period (1s each):
+
+| Period         | Requests within period                         | Tokens consumed |
+|----------------|-------------------------------------------------|-----------------|
+| 0–1000 ms      | 0ms:7, 200ms:5, 650ms:3                        | 15              |
+| 1000–2000 ms   | 1200ms:6, 1800ms:5                             | 11              |
+| 2000–3000 ms   | 2100ms:10 (insufficient), 2600ms:10 (consumed) | 10              |
+
+Total consumed over 3 seconds: 15 + 11 + 10 = 36 tokens.
+
+## Old Algorithm (Pre-unification) — Scenario 1 Timeline
```

Comment:
this section of the docs really belongs in the PR - ie it's probably prudent to refactor this document and PR comment so that this document becomes the PR comment and the _parts of_ the doc that explain the new behavior become the documentation - "how it used to work" is hardly interesting for anyone looking to understand to use the token bucket now. It is only interesting when porting code from one version to another, for which the PR / commit message are better sources.